### PR TITLE
Adding Build Workflow 

### PR DESCRIPTION
**Pull request by shooshmand-sol**
### What is the purpose of this change?
Adding a build workflow, currently it runs PMD/Checkstile/Findbugs/Unit Tests
Buidling the Jar file, SonarQube, Whitesource will be added later

### How was this change implemented?
Adding a build workflow to get triggered when we merge into main branch and for PRs

### How was this change tested?
Sample Run: 
https://github.com/SolaceLabs/event-management-agent/actions/runs/4865219741/jobs/8675255781

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: 'temurin'
+          cache: 'maven'
+      - name: Static Code Analysis
+        run: mvn -B compile -Pmaas-static-code-analysis --file service/pom.xml
+      - name: Unit Tests
+        run: mvn -B test -Dcheckstyle.skip -Dpmd.skip -Dcpd.skip -Dfindbugs.skip -Dspotbugs.skip --file service/pom.xml
```
Comment (shooshmand-sol) [inline .github/workflows/build.yaml:25]: we do not want to re-run PMD/findbugs test again in this step, since they have been executed in previous step.

Comment (amanek-solace) [reply .github/workflows/build.yaml:25]: we don't need to to save some cycles in the build but if i remember correctly the sca tools doesn't take a long time to execute.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Build
+on:
+  push:
+    branches:
```
Comment (johnvincentcorpuz) [inline .github/workflows/build.yaml:4]: This is a good first step. But would recommend adding exceptions for builds/tests to run on specific paths: i.e https://github.com/SolaceDev/maas-billing-v2/blob/8961d14770a4113486267a95cea288e51009ff3e/.github/workflows/release.yml#L5

Comment (johnvincentcorpuz) [reply .github/workflows/build.yaml:4]: Or only run build/test when there's a change in the service folder. Example: https://github.com/SolaceDev/maas-billing-v2/blob/8961d14770a4113486267a95cea288e51009ff3e/.github/workflows/build-helm-chart.yml#L10

File: .github/workflows/build.yaml (hunk 1 above)
Comment (shooshmand-sol) [reply .github/workflows/build.yaml:25]: But I'm ok to remove the skip stages.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (shooshmand-sol) [reply .github/workflows/build.yaml:4]: That's a good idea, so we dont need to run tests on for exampl egithub actions changes.

Comment (shooshmand-sol) [reply .github/workflows/build.yaml:4]: after discussing it with @amanek-solace we decided to add this later.

Comment (johnvincentcorpuz) [reply .github/workflows/build.yaml:4]: lol that's a two line change: 
```
  paths:
      - 'service/**'
```

**Pull request merged**